Lib/email/utils.py

```diff
@@ -106,12 +106,62 @@ def formataddr(pair, charset='utf-8'):
     return address
 
 
+def _pre_parse_validation(email_header_fields):
+    accepted_values = []
+    for v in email_header_fields:
+        s = v.replace('\\(', '').replace('\\)', '')
+        if s.count('(') != s.count(')'):
```

Comment:
This does not check `)(` or any other combination when `)` proceeds `(`. Probably it could be better to iterate through the string and manually increment and decrement verifying that the counter never goes below `0` and it is exactly `0` at the end
```suggestion
        counter = 0
        for c in s:
            if c == '(':
                counter += 1
            elif c == ')':
                counter -= 1
                if counter < 0:
                    break
        if counter != 0:
```

Reply:
Looks like this is not necessary. The fix already accounts for this and none of those are an issue.

```
./python
Python 3.13.0a0 (heads/blah_1:a1cc74c4ee, Aug 21 2023, 18:40:57) [GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> from email.utils import parseaddr, getaddresses
>>> 
>>> 
>>> parseaddr('alice@example.org])(<bob@example.com>')
('', '')
>>> 
>>> parseaddr('alice@example.org)(<bob@example.com>')
('', '')
>>> 
>>> parseaddr('alice@example.org <bob@example.com>')
('', '')
>>> parseaddr('alice@example.org..<bob@example.com>')
('', '')
>>> 
>>> getaddresses(['alice@example.org..<bob@example.com>'])
[('', '')]
>>> 
>>> getaddresses(['alice@example.org])(<bob@example.com>'])
[('', '')]
>>> 
>>> getaddresses(['alice@example.org)(<bob@example.com>'])
[('', '')]
```

Reply:
After looking at this again, it is interesting that my solution works here… why do I check for matching parentheses at all ?

`blah@example.com)(<bob&@example.com`

I have ideas, but I’ll look into it.